packages/markers/src/browser/markers-tree.ts

```diff
@@ -47,53 +44,55 @@ export class MarkersTree extends Tree {
     }
 
     getMarkerInfoNodes(parent: MarkerRootNode): Promise<MarkerInfoNode[]> {
-        const markerFiles = this.markersManager.getMarkerInformationByKind(this.markerRoot.kind);
         const uriNodes: MarkerInfoNode[] = [];
-        markerFiles.forEach(markerInfo => {
-            const cachedMarkerInfo = this.markerInfoNodeCache.find(m => m.uri.toString() === markerInfo.uri);
-            const markerInfoNodeUri = new URI(markerInfo.uri);
-            const markerInfoNode: MarkerInfoNode = {
-                children: [],
-                expanded: cachedMarkerInfo ? cachedMarkerInfo.expanded : false,
-                uri: markerInfoNodeUri,
-                id: 'markerInfo-' + markerInfoNodeUri.displayName,
-                name: markerInfoNodeUri.displayName,
-                parent,
-                selected: false,
-                numberOfMarkers: markerInfo.counter
-            };
-            uriNodes.push(markerInfoNode);
+        this.markersManager.forEachMarkerInfoByKind(this.markerRoot.kind, markerInfo => {
+            const uri = new URI(markerInfo.uri);
+            const id = 'markerInfo-' + markerInfo.uri;
+            const cachedMarkerInfo = this.getNode(id);
+            if (cachedMarkerInfo && MarkerInfoNode.is(cachedMarkerInfo)) {
+                uriNodes.push(cachedMarkerInfo);
```

Comment:
Should not you update the existing node with new counts?

Reply:
Numbers are totally overrated...but for you... ;-)
https://github.com/theia-ide/theia/pull/455/commits/ac0e1e94cab4a13848f308ad6c3a9374341cb518#diff-b17954869c4dc749299bb098b8a07c92R53